schema.graphql

```diff
@@ -94,3 +94,59 @@ type Settle @entity(immutable: true) {
   blockTimestamp: BigInt!
   transactionHash: Bytes!
 }
+
+type Challenge @entity {
+  id: Bytes! # uint256
+  verifier: Bytes! # address
+  minimumCheckIns: BigInt! # uint64
+  startTimestamp: BigInt! # uint64
+  joinDueTimestamp: BigInt! # uint64
+  endTimestamp: BigInt! # uint64
+  donateDestination: Bytes! # address
+  checkInJudge: Bytes! # address
+  asset: Bytes! # address
+  donationBPS: BigInt! # uint128
+  stakePerUser: BigInt! # uint128
+  status: BigInt! # uint8
+  joinedUsers: [UserChallenge!] @derivedFrom(field: "challengeId") # User[]
+  totalUsers: BigInt! # uint256
+  totalStake: BigInt! # uint256
+  totalCheckIns: BigInt! # uint256
+  totalClaims: BigInt! # uint256
+  totalSucceedUsers: BigInt! # uint256
+  totalFailedUsers: BigInt! # uint256
+  totalDonation: BigInt #TODO uint256
+}
+
+type User @entity {
+  id: Bytes! # address
+  joinedChallenges: [UserChallenge!] @derivedFrom(field: "user") # Challenge[]
```

Comment:
we can combined `joinedChallenges`, `claimedChallenges`, `settledChallenges` into as single `history: [UserChallenge!]`

or is it something you plan on doing?

Reply:
I think it's the naming issue, I've renamed it to make it clearer. I want to make joinedChallenges -> history. And claimedChallenges and settledChallenges is more like the personal record of the claiming history and the challenges that the user help settled. Does that make sense to you?